### PR TITLE
luau: add version 0.548 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **luau/***

Since luau/0.548, Luau.CodeGen depends Luau.VM and LuaU.VM.Internals.
So conan recipe has to fix LUAI_FUNC definition.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/luau/all/conanfile.py [hunk 1]
```diff
@@ -25,72 +30,80 @@ class LuauConan(ConanFile):
...
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
```
Comment (jwillikers) [inline recipes/luau/all/conanfile.py:41]: Can this be removed since the `check_min_vs` call exists below?

File: recipes/luau/all/conanfile.py [hunk 2]
```diff
@@ -25,72 +30,80 @@ class LuauConan(ConanFile):
...
+        check_min_vs(self, 191)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support."
+            )
 
         if is_msvc(self) and self.options.shared:
-            raise tools.ConanInvalidConfiguration("{} does not support shared build in MSVC".format(self.name))
+            raise ConanInvalidConfiguration("{} does not support shared build in MSVC".format(self.name))
```
Comment (jwillikers) [inline recipes/luau/all/conanfile.py:76]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} does not support shared build in MSVC")
```

File: recipes/luau/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,11 +1,8 @@
 cmake_minimum_required(VERSION 3.8)
 project(test_package)
```
Comment (jwillikers) [inline recipes/luau/all/test_package/CMakeLists.txt:2]:
```suggestion
project(test_package LANGUAGES CXX)
```

File: recipes/luau/all/test_v1_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package)
```
Comment (jwillikers) [inline recipes/luau/all/test_v1_package/CMakeLists.txt:2]:
```suggestion
project(test_package LANGUAGES CXX)
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/luau/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/luau/all/conanfile.py:41]: I agree with you.
It's my fault.

File: recipes/luau/all/test_v1_package/CMakeLists.txt (hunk 4 above)
Comment (toge) [reply recipes/luau/all/test_v1_package/CMakeLists.txt]: fixed

File: recipes/luau/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (toge) [reply recipes/luau/all/test_package/CMakeLists.txt]: fixed

File: recipes/luau/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/luau/all/conanfile.py]: fixed

File: recipes/luau/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/luau/all/conanfile.py]: fixed

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**